### PR TITLE
Memory Challenge C

### PR DESCRIPTION
**Pull request by dschonholtz**
### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: tests/integration/challenges/memory/test_memory_challenge_c.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import pytest
```
Comment (waynehamadi) [inline tests/integration/challenges/memory/test_memory_challenge_c.py:1]: Great !
Could you update the mkdocs.yml
```
- Memory:
    - Introduction: challenges/memory/introduction.md
    - Memory Challenge A: challenges/memory/challenge_a.md
    - Memory Challenge B: challenges/memory/challenge_b.md
    - Memory Challenge C: challenges/memory/challenge_c.md
```

And then put a description of the challenges in memory/challenge_c.md (you can copy memory/challenge_b.md and modify it)

**Review by merwanehamadi — CHANGES_REQUESTED** [review]

File: tests/integration/challenges/memory/test_memory_challenge_c.py (hunk 1 above)
Comment (dschonholtz) [reply tests/integration/challenges/memory/test_memory_challenge_c.py:1]: Done!

**Comment by merwanehamadi** [comment]
@dschonholtz sorry almost there: 
black .
and isort .
and then the `in the wiki (click on vercel above to see the preview)
Screenshot 2023-05-06 at 8 35 54 PM

**Comment by dschonholtz** [comment]
The memory challenge c is registering as failed here even though I run it locally with the cassette fine. I deleted the cassette and re-ran it locally where it passed fine again and I pushed the new cassette up. 
It appears to be trying to run the test without an API key in the cloud ignoring the cassette.
I'm guessing it is registering as different because I ran the test by itself? I am re-running one more time with pytest for the whole repo after deleting the cassette again locally and going to try to re-push. 
If you know what is going on that'd be neat

**Review by merwanehamadi — APPROVED** [review]

**Pull request merged**